src/parser/mod.rs

```diff
@@ -6936,6 +6942,162 @@ impl<'a> Parser<'a> {
             having,
             named_window: named_windows,
             qualify,
+            from_before_select: false,
+        })
+    }
+
+    /// Parse a duckdb style `FROM` statement without a select.
+    /// assuming the initial `FROM` was already consumed
+    pub fn parse_from_select(&mut self) -> Result<Select, ParserError> {
```

Comment:
this seems largely copied from the normal select parsing -- is there any way to refactor the code to reduce the duplication